internal/controller/storageclient_controller.go

```diff
@@ -329,6 +348,63 @@ func (r *storageClientReconcile) reconcileDynamicWatches() error {
 	return nil
 }
 
+func (r *storageClientReconcile) reconcileOdfVolumeGroupSnapshot() error {
+	if watchExists, foundCrd := r.crdsBeingWatched.Load(OdfVolumeGroupSnapshotClassCrdName); !foundCrd || watchExists.(bool) {
+		return nil
+	}
+
+	crd := &metav1.PartialObjectMetadata{}
+	crd.SetGroupVersionKind(extv1.SchemeGroupVersion.WithKind("CustomResourceDefinition"))
+	crd.Name = OdfVolumeGroupSnapshotClassCrdName
+	if err := r.get(crd); client.IgnoreNotFound(err) != nil {
+		return err
+	}
+	// CRD doesn't exist in the cluster
+	if crd.UID == "" {
+		r.log.Info("ODF-VolumeGroupSnapshotClass CRD is not available", "crdName", OdfVolumeGroupSnapshotClassCrdName)
```

Comment:
doesn't add much value. if the crd doesn't exist we anyways get a log when we try to reconcile resource of that kind already.